File: lib/ts/src/internal/types/TBoolean.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { ValidationFailure } from '../validation/ValidationFailure.js';
import { TTypeDeclaration } from './TTypeDeclaration.js';
import { TType } from './TType.js';
import { validateBoolean } from '../validation/ValidateBoolean.js';
import { generateRandomBoolean } from '../generation/GenerateRandomBoolean.js';
import { GenerateContext } from '../generation/GenerateContext.js';
import { ValidateContext } from '../validation/ValidateContext.js';

export const booleanName: string = 'Boolean';

export class TBoolean extends TType {
    getTypeParameterCount(): number {
        return 0;
    }

    validate(value: any, typeParameters: TTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return validateBoolean(value);
    }

    generateRandomValue(
        blueprintValue: any,
        useBlueprintValue: boolean,
        typeParameters: TTypeDeclaration[],
        ctx: GenerateContext,
    ): any {
        return generateRandomBoolean(blueprintValue, useBlueprintValue, ctx);
    }

    getName(): string {
        return booleanName;
    }
}

File: lib/go/internal/types/TBoolean.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

const booleanName = "Boolean"

// TBoolean represents the Telepact boolean type.
type TBoolean struct{}

// NewTBoolean constructs a TBoolean instance.
func NewTBoolean() *TBoolean {
	return &TBoolean{}
}

// GetTypeParameterCount implements TType.
func (t *TBoolean) GetTypeParameterCount() int {
	return 0
}

// Validate implements TType.
func (t *TBoolean) Validate(value any, typeParameters []*TTypeDeclaration, ctx *ValidateContext) []*ValidationFailure {
	return ValidateBoolean(value)
}

// GenerateRandomValue implements TType.
func (t *TBoolean) GenerateRandomValue(blueprintValue any, useBlueprintValue bool, typeParameters []*TTypeDeclaration, ctx *GenerateContext) any {
	return GenerateRandomBoolean(blueprintValue, useBlueprintValue, ctx)
}

// GetName implements TType.
func (t *TBoolean) GetName(ctx *ValidateContext) string {
	return booleanName
}
